lib/project-overlays.nix

```diff
@@ -0,0 +1,114 @@
+{
+  lib
+, haskellLib
+}: {
+  devshell = final: prev: {
```

Comment:
what does this do? is it a competitor to `shellFor`? Any explanation at all? Comments?

Reply:
That's pretty non-obvious... do we want to support a hook into some other framework indefinitely in haskell.nix?

Reply:
I think it is useful to have code to make it easier to use Haskell.nix with different frameworks and the like in Haskell.nix.  If others find a use for them it will make it more likely that people will use and contribute to Haskell.nix.

It would be nice though if the cabal wrapper could be extracted so that it could be used with plain `shellFor` code as well.

Reply:
I actually really disagree. I think the cabal wrapper is a hack and we shouldn't propagate it. Hopefully we'll be able to bin it for good once we're using a custom hackage.

Generally I still think we have way too much stuff in haskell.nix, and so the bar for adding new stuff should be much higher...